crates/red_knot_module_resolver/src/path.rs

```diff
@@ -725,6 +725,82 @@ impl PartialEq<ModuleResolutionPathRef<'_>> for VendoredPathBuf {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub(crate) struct ModuleSearchPath(Arc<ModuleResolutionPathBuf>);
```

Comment:
The naming is kind of confusing now with `ModuleSearchPath`, `ModuleSearchPathBuf` and `ModuleSearchPathRef`. 

Can we rename the `PathBuf` and `PathRef` structs to maybe `ModulePathBuf` and `ModulePathRef`. At least I understand that these are now paths to modules and not to the search paths. 

These structs could also use some documentation to explain the intent.

Reply:
> The naming is kind of confusing now with `ModuleSearchPath`, `ModuleSearchPathBuf` and `ModuleSearchPathRef`.

Those aren't the names I've used -- the three structs are `ModuleSearchPath`, `ModuleResolutionPathBuf` and `ModuleResolutionPathRef`, which isn't _so_ confusing. But I like your proposed rename.

> These structs could also use some documentation to explain the intent.

Sure

Reply:
I also think the word "Resolution" doesn't add value, and would prefer the names Micha suggested.